src/app/shared/image/image-loader.component.ts

```diff
@@ -1,21 +1,42 @@
 import { Component, Input, ElementRef } from '@angular/core';
+import { HalDoc } from '../cms/haldoc';
 
 @Component({
   selector: 'image-loader',
   styleUrls: ['image-loader.component.css'],
   template: `
-    <img *ngIf="src" [src]="src" class="loading" (load)="onLoad()"/>
+    <img *ngIf="src" [src]="src" (load)="onLoad()" (error)="onError()"/>
     `
 })
 
 export class ImageLoaderComponent {
 
   @Input() public src: string;
+  @Input() public imageDoc: HalDoc;
```

Comment:
Also, if I'm reading this right, you can still just pass in a `src` right?  It just has to be non-null on init, or you'll get the placeholder image.